.github/templates/odfe-testing-cluster-template.json

```diff
@@ -0,0 +1,238 @@
+            "Description": "Security feature enabled"
+        },
+        "vpcId":{
+            "Type": "String",
+            "Description": "VPC id associated with the cluster"
+        },
+
+        "ODFESecurityGroup":{
+            "Type" : "String",
+            "Description": "Security Group id to be attached to all the resources"
+        },
+        "keypair":{
+            "Type" : "String",
+            "Description": "Security Group id to be attached to all the resources"
+        }
+  
+    },
+
+    "Mappings" : {
+        "DistributionMap" : {
+          "RPM" : { "amiId" : "ami-0e34e7b9ca0ace12d" },
+          "TAR" : { "amiId" : "ami-003634241a8fcdec0" },
+          "DEB" : { "amiId" : "ami-003634241a8fcdec0" }
+        }
+      },
+
+    "Conditions" : {
+        "CreateSecurityResources" : {"Fn::Equals" : [{"Ref" : "security"}, "ENABLE"]},
+        "CreateDisableSecurityResources" : {"Fn::Equals" : [{"Ref" : "security"}, "DISABLE"]}
+    },
+  
+    "Resources": {         
+  
+        "ODFEASG" : {
+            "Type" : "AWS::AutoScaling::AutoScalingGroup",
+            "Properties" : {
+                "AvailabilityZones" : [ "us-west-2a","us-west-2b","us-west-2c","us-west-2d"  ],
+                "LaunchConfigurationName" : { "Ref" : "asgLaunchConfig" },
+                "MinSize" : "1",
+                "MaxSize" : "1",
+                "DesiredCapacity" : "1",
+                "TargetGroupARNs": [{"Fn::If": ["CreateSecurityResources", {"Ref" : "ESSTargetGroup"}, {"Ref":"ESTargetGroup"}]}, 
+                        {"Ref":"KibanaTargetGroup"}],
+                "Tags":[{
+                    "Key" : "Name", 
+                    "Value" : {
+                        "Fn::Join": [
+                          "", [
+                            "ODFE-",
+                            {
+                              "Ref": "distribution"
+                            },
+                            "-SECURITY-",
+                            {
+                                "Ref": "security"
+                            },
+                            "-Testing-Cluster"
+                          ]
+                        ]
+                      }, 
+                    "PropagateAtLaunch" : "true"}]
+                }
+            },
+
+        "asgLaunchConfig":{
+              "Type": "AWS::AutoScaling::LaunchConfiguration",
+              "Properties": {
+                  "ImageId": { "Fn::FindInMap" : [ "DistributionMap", { "Ref" : "distribution" }, "amiId"]},
+                  "InstanceType": "m5a.large",
+                  "IamInstanceProfile": "odfe_testing_cluster_role",
+                  "KeyName": {"Ref" : "keypair"},
+                  "AssociatePublicIpAddress": true,
+                  "SecurityGroups": [{"Ref" : "ODFESecurityGroup"}],
+                  "UserData": {
+                      "Ref": "userdata"}
+                  
+                }
+        },
+
+
+        "ESLoadBalancer" : {
```

Comment:
If we create the ELB for each request, then for one release, the domain name will keep changing. To avoid this, it's better to create the ELB separately.

Reply:
> If we create the ELB for each request, then for one release, the domain name will keep changing. To avoid this, it's better to create the ELB separately.

You mean keep ELB's ready and just map them with each request?

Reply:
yes. that's right.

Reply:
That would need some additional work. Will work on it. Right now this script creates new ELB for each request but as you said if we rerun the same request (eg: create rpm auth cluster and then recreate rpm auth cluster again) the APIs will be different.

Reply:
Let's raise another PR for this change